elastic/apm-server
# processor/otel: index span events as logs

## Motivation/summary

Produce log events for non-exception OpenTelemetry span events, capturing the span event name as the log message, and all other attributes as labels. We only index logs when data streams are enabled; they are dropped when classic indices are in use.

## Checklist


## How to test these changes

1. Instrument an application with either an OpenTelemetry SDK or Jaeger, capturing a span and span logs/events.
2. Run APM Server with data streams enabled, check that logs are indexed into `logs-apm.app-<namespace>` and show up in the "Logs" panel for a trace sample
3. Run APM Server with data streams disabled, check that logs are not indexed at all

## Related issues

Closes https://github.com/elastic/apm-server/issues/4715
Closes https://github.com/elastic/apm-server/issues/3338
